cpp/src/DataStorm/ConnectionManager.cpp

```diff
@@ -12,21 +15,20 @@ ConnectionManager::ConnectionManager(const shared_ptr<CallbackExecutor>& executo
 void
 ConnectionManager::add(
     const shared_ptr<void>& object,
-    const shared_ptr<Ice::Connection>& connection,
-    function<void(const shared_ptr<Ice::Connection>&, exception_ptr)> callback)
+    const Ice::ConnectionPtr& connection,
+    function<void(const Ice::ConnectionPtr&, exception_ptr)> callback)
 {
     lock_guard<mutex> lock(_mutex);
     auto& objects = _connections[connection];
     if (objects.empty())
     {
-        connection->setCloseCallback([self = shared_from_this()](const shared_ptr<Ice::Connection>& con)
-                                     { self->remove(con); });
+        connection->setCloseCallback([self = shared_from_this()](const Ice::ConnectionPtr& con) { self->remove(con); });
     }
-    objects.emplace(move(object), move(callback));
+    objects.emplace(std::move(object), std::move(callback));
```

Comment:
the std::move on the `const shared_ptr&` does not look correct.

Reply:
You are right, I updated the code to pass it by value, also changed the connection to be the first parameter.